app/models/user.rb

```diff
@@ -173,6 +175,14 @@ class User < ApplicationRecord
   # minimum number of submitted stories before checking self promotion
   MIN_STORIES_CHECK_SELF_PROMOTION = 2
 
+  def underscores_and_dashes_in_username
+    username = self.username.split(/_|-/)
+    return if username.length < 2
+
+    collisions = User.where('username REGEXP ?', username.join("[-_]"))
```

Comment:
Clever approach!

To build on it, you could combine your `split` to express this as `username.gsub(/_|-/, "[-_]")`.